docs/DeveloperGuide.md

```diff
@@ -1,38 +1,575 @@
 # Developer Guide
 
 ## Acknowledgements
+We used these third party libraries to develop our application:
+- Gson
+- Mockito
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+## Design
 
-## Design & implementation
+### DailyRecord component
+API: `DailyRecord.java`
+![Diagram for DailyRecord Component](./images/DailyRecord_API_UML.jpg)
```

Comment:
You could omit methods that are not directly relevant. Such as the toString() method in the DailyRecord UML diagram

Reply:
Navigability is missing from Day, MealList and Water class

Reply:
![Uploading image.png…]()